vowpalwabbit/parse_example_json.h

```diff
@@ -1668,13 +1668,15 @@ void read_line_json(vw& all, v_array<example*>& examples, char* line, example_fa
 
 inline void apply_pdrop(vw& all, float pdrop, v_array<example*>& examples)
 {
+  // Event with certain pdrop had (1-pdrop) as probability to survive, 
+  // so it is one of (1 / (1-pdrop)) events that we should learn on, and weight should be updated accordingly.
```

Comment:
doesn't this imply that there are `1-pdrop` events so each one has a probability of `1/(1-pdrop)`?

should we multiplying with `1-pdrop` instead of `1/(1-pdrop)`?

Reply:
ok I understand we want higher impact (weight) for higher pdrop which is why we are dividing here (maybe the comment got me confused)

different concern: if pdrop is 1 we are dividing by zero here

Reply:
yeah that is true. updated.

Reply:
why not set the weight to 0, wouldnt that be equivalent to dropping all events, pdrop of 1

Reply:
it may work, but could also be some corner case (if there is division by weight somewhere), which has to be extra tested.